src/jit/regset.cpp

```diff
@@ -3320,6 +3320,14 @@ void Compiler::tmpPreAllocateTemps(var_types type, unsigned count)
         tmpCount++;
         tmpSize += size;
 
+#ifdef _TARGET_ARM_
+        if (type == TYP_DOUBLE)
+        {
+            // Adjust tmpSize in case it needs alignment
+            tmpSize += TARGET_POINTER_SIZE;
```

Comment:
This looks like it's a port from the `LEGACY_BACKEND` code for ARM32 in `tmpGetTemp`. Is that correct?

I would be interested in better understanding the behavior here as well.

cc @CarolEidt who may have some additional insight into codegen temp management.